src/train_model.py

```diff
@@ -127,26 +127,27 @@ def train_model(config: Struct):
         f"{-torch.log(torch.tensor(1 / config.vocab_size))}")
 
     # Checks to see if you want to use a portion of the dataset for training
-    if config.split_dataset:
-        print(f"\nUsing {config.split_dataset_percentage}% of the dataset for training")
+    if (config.split_dataset != 1.0):
+        print(f"\nUsing {config.split_dataset * 100}% of the dataset for training")
 
         # Define the source and target directories
-        source_dir = Path("/grphome/grp_retnet/compute/data/datasets_tokenized/c4/train")
-        target_dir = Path("/grphome/grp_retnet/compute/data/datasets_tokenized/c4/portion_data")
+        source_dir = Path(config.tokenized_dataset_path + "/train")
+        target_dir = Path(config.tokenized_dataset_path + "/portion_data")
 
         # Ensure the target directory exists
         target_dir.mkdir(parents=True, exist_ok=True)
 
         # Calculate the number of files to copy based on the percentage
-        total_files = 1031  # Total number of parquet files in the source directory
-        files_to_copy = int((config.split_dataset_percentage / 100) * total_files)
+        total_files = len(list(source_dir.glob('*.parquet')))
+        files_to_copy = int((config.split_dataset) * total_files)
 
         # Count the existing files in the target directory
         existing_files_count = len(list(target_dir.glob('*.parquet')))
+        print(f"\nThere are {files_to_copy} files being copied which is actually {(files_to_copy / existing_files_count)*100}% of the training data")
```

Comment:
`There are {existing_files_count} .parquet files available, of which {files_to_copy} files are being copied which is actually {(files_to_copy / existing_files_count)*100}% of the training data, instead of {config.split_dataset * 100}%` would be a better print statement